runtime/src/rpc.rs

```diff
@@ -220,8 +219,18 @@ impl InterBtcParachain {
             .map(|x| x.nonce)
             .unwrap_or_default();
 
-        log::info!("Refreshing nonce: {}", account_info);
-        signer.set_nonce(account_info);
+        let mut next_nonce = self.nonce.write().await;
+
+        let ret = if on_chain_nonce > *next_nonce {
+            log::info!("Synced to on-chain nonce: {}", on_chain_nonce);
+            on_chain_nonce
+        } else {
+            *next_nonce
+        };
+
+        *next_nonce = next_nonce.saturating_add(1);
```

Comment:
That was indeed the intention, good spot!

Reply:
@sander2 is `self.nonce` actually used anywhere? I'm wondering why it was added in the first place

Reply:
It's fetched & updated right in this function? All other functions should be using this function. This change was because the nonce used to be embedded in the signer, but subxt refactored it such that the nonce is now an explicit argument to be used in `create_signed_with_nonce`

Reply:
Also if you're asking why we cache the nonce at all, it's because when you make 2 concurrent calls, you need both to be signed with unique nonces

Reply:
In that case I think the `self.nonce` field is used purely for the mutex property rather than as a cache, maybe something to change later